iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string_internal.hpp

```diff
@@ -93,8 +98,9 @@ struct GetSize<std::string>
 template <>
 struct GetSize<char>
 {
-    static uint64_t call(char) noexcept
+    static uint64_t call(char c) noexcept
     {
+        IOX_DISCARD_RESULT(c);
```

Comment:
```suggestion
    static uint64_t call(char IOX_MAYBE_UNUSED(c)) noexcept
    {
```
I think the code above should also work.